NPE at Eclipse startup

If i have a m2e-android handled project in my workspace, i always get this stacktrace at startup:
```
!ENTRY org.eclipse.core.resources 4 2 2015-04-21 00:08:04.207
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.core.resources".
!STACK 0
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2199)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3932)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3936)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4806)
	at me.gladwell.eclipse.m2e.android.configuration.CachingClasspathLoader.load(CachingClasspathLoader.java:53)
	at me.gladwell.eclipse.m2e.android.configuration.NonRuntimeDependenciesClasspathContainer.getClasspathEntries(NonRuntimeDependenciesClasspathContainer.java:39)
	at org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:2699)
	at org.eclipse.jdt.internal.core.JavaProject.resolveClasspath(JavaProject.java:2857)
	at org.eclipse.jdt.internal.core.JavaProject.getResolvedClasspath(JavaProject.java:1962)
	at org.eclipse.jdt.internal.core.ExternalFoldersManager.refreshReferences(ExternalFoldersManager.java:391)
	at org.eclipse.jdt.internal.core.DeltaProcessor.resourceChanged(DeltaProcessor.java:2029)
	at org.eclipse.jdt.internal.core.DeltaProcessingState.resourceChanged(DeltaProcessingState.java:477)
	at org.eclipse.core.internal.events.NotificationManager$1.run(NotificationManager.java:291)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.NotificationManager.notify(NotificationManager.java:285)
	at org.eclipse.core.internal.events.NotificationManager.handleEvent(NotificationManager.java:267)
	at org.eclipse.core.internal.resources.Workspace.broadcastEvent(Workspace.java:373)
	at org.eclipse.core.internal.resources.Resource.refreshLocal(Resource.java:1704)
	at com.android.ide.eclipse.adt.internal.project.BaseProjectHelper.markProject(BaseProjectHelper.java:220)
	at com.android.ide.eclipse.adt.internal.project.BaseClasspathContainerInitializer$1.run(BaseClasspathContainerInitializer.java:60)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)
Caused by: java.lang.NullPointerException
	at me.gladwell.eclipse.m2e.android.configuration.ClasspathAttacher.attach(ClasspathAttacher.java:55)
	at me.gladwell.eclipse.m2e.android.configuration.AttachDocumentsClasspathLoader.load(AttachDocumentsClasspathLoader.java:36)
	at me.gladwell.eclipse.m2e.android.configuration.ClasspathLoaderDecorator.load(ClasspathLoaderDecorator.java:26)
	at me.gladwell.eclipse.m2e.android.configuration.AttachSourcesClasspathLoader.load(AttachSourcesClasspathLoader.java:33)
	at me.gladwell.eclipse.m2e.android.configuration.CachingClasspathLoader$1.load(CachingClasspathLoader.java:39)
	at me.gladwell.eclipse.m2e.android.configuration.CachingClasspathLoader$1.load(CachingClasspathLoader.java:1)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3522)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2315)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2278)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2193)
	... 20 more
```

Using the latest snapshot m2e-android, ADT, Eclipse Luna.

--- a/me.gladwell.eclipse.m2e.android/src/main/java/me/gladwell/eclipse/m2e/android/configuration/ClasspathAttacher.java
+++ b/me.gladwell.eclipse.m2e.android/src/main/java/me/gladwell/eclipse/m2e/android/configuration/ClasspathAttacher.java
@@ -27,6 +27,7 @@ import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.jdt.core.IClasspathEntry;
 import org.eclipse.jdt.core.IJavaProject;
 import org.eclipse.m2e.core.embedder.IMaven;
+import org.eclipse.m2e.core.project.IMavenProjectFacade;
 import org.eclipse.m2e.core.project.IMavenProjectRegistry;
 
 import com.google.inject.Inject;
@@ -52,7 +53,14 @@ public class ClasspathAttacher<T extends EntryAttacher> {
     public Iterable<IClasspathEntry> attach(IJavaProject project, Iterable<IClasspathEntry> classpath) {
         List<IClasspathEntry> processed = new ArrayList<IClasspathEntry>();
         try {
-            MavenProject mavenProject = registry.getProject(project.getProject()).getMavenProject(new NullProgressMonitor());
+            IMavenProjectFacade mavenProjectFacade = registry.getProject(project.getProject());
+
+            if (mavenProjectFacade == null) {
+                warn("maven project not yet registered for " + project);
+                return processed;
+            }
+
+            MavenProject mavenProject = mavenProjectFacade.getMavenProject(new NullProgressMonitor());
             MavenAndroidProject androidProject = factory.createAndroidProject(mavenProject);
             List<ArtifactRepository> repositories = mavenProject.getRemoteArtifactRepositories();
 

